tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py

```diff
@@ -463,6 +466,12 @@ def _reset_ids(self, ids):
       s._reset_ids(ids)
 
 
+# TODO (Lifann) Introduce ShadowVariable when using tf.function.
+# Could hack the tf.function and mark whether if the Python context
+# is in the `function` scope. When inside `function` scope, create
+# ShadowVariable out of the scope, and then do the lookup. Also
```

Comment:
“Shadow is a persistant twin of TrainableWrapper.”  maybe more clear.

Reply:
This is already implemented yet. I'll move it into annotations.